File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/ui/utils/PopUpActivityPresenter.java
package com.lattis.ellipse.presentation.ui.utils;

import android.os.Bundle;
import androidx.annotation.Nullable;

import com.lattis.ellipse.presentation.ui.base.activity.ActivityPresenter;

import javax.inject.Inject;

import static com.lattis.ellipse.presentation.ui.utils.PopUpActivity.ACTIONBTN_POP_UP;
import static com.lattis.ellipse.presentation.ui.utils.PopUpActivity.SUBTITLE1_POP_UP;
import static com.lattis.ellipse.presentation.ui.utils.PopUpActivity.SUBTITLE2_POP_UP;
import static com.lattis.ellipse.presentation.ui.utils.PopUpActivity.TITLE_POP_UP;

/**
 * Created by ssd3 on 4/26/17.
 */

public class PopUpActivityPresenter extends ActivityPresenter<PopUpActivityView> {

    private String title=null;
    private String subTitle1=null;
    private String subTitle2=null;
    private String actionBtn=null;


    @Inject
    PopUpActivityPresenter(){

    }

    @Override
    protected void setup(@Nullable Bundle arguments) {
        super.setup(arguments);
        if (arguments != null) {
            if (arguments.containsKey(TITLE_POP_UP)) {
                title = arguments.getString(TITLE_POP_UP);
            }
            if (arguments.containsKey(SUBTITLE1_POP_UP)) {
                subTitle1 = arguments.getString(SUBTITLE1_POP_UP);
            }
            if (arguments.containsKey(SUBTITLE2_POP_UP)) {
                subTitle2 = arguments.getString(SUBTITLE2_POP_UP);
            }
            if (arguments.containsKey(ACTIONBTN_POP_UP)) {
                actionBtn = arguments.getString(ACTIONBTN_POP_UP);
            }
            view.setView(title,subTitle1,subTitle2,actionBtn);
        }
    }

    @Override
    protected void updateViewState() {

    }
}

File: velolabs_repos/operator-android/app/src/main/java/io/lattis/operator/presentation/popup/PopUpActivityPresenter.kt
package io.lattis.operator.presentation.popup

import android.os.Bundle
import io.lattis.operator.presentation.base.activity.ActivityPresenter
import javax.inject.Inject

class PopUpActivityPresenter @Inject internal constructor() :
    ActivityPresenter<PopUpActivityView>() {
    private var title: String? = null
    private var subTitle1: String? = null
    private var subTitle2: String? = null
    private var actionBtnPositive1: String? = null
    private var actionBtnPositive2: String? = null
    private var actionBtnPositive3: String? = null
    private var actionBtnNegative: String? = null

    override fun setup(arguments: Bundle?) {
        super.setup(arguments)
        if (arguments != null) {
            if (arguments.containsKey(PopUpActivity.TITLE_POP_UP)) {
                title = arguments.getString(PopUpActivity.TITLE_POP_UP)
            }
            if (arguments.containsKey(PopUpActivity.SUBTITLE1_POP_UP)) {
                subTitle1 = arguments.getString(PopUpActivity.SUBTITLE1_POP_UP)
            }
            if (arguments.containsKey(PopUpActivity.SUBTITLE2_POP_UP)) {
                subTitle2 = arguments.getString(PopUpActivity.SUBTITLE2_POP_UP)
            }
            if (arguments.containsKey(PopUpActivity.ACTIONBTN_POSITIVE1)) {
                actionBtnPositive1 = arguments.getString(PopUpActivity.ACTIONBTN_POSITIVE1)
            }
            if (arguments.containsKey(PopUpActivity.ACTIONBTN_POSITIVE2)) {
                actionBtnPositive2 = arguments.getString(PopUpActivity.ACTIONBTN_POSITIVE2)
            }
            if (arguments.containsKey(PopUpActivity.ACTIONBTN_POSITIVE3)) {
                actionBtnPositive3 = arguments.getString(PopUpActivity.ACTIONBTN_POSITIVE3)
            }
            if (arguments.containsKey(PopUpActivity.ACTIONBTN_NEGATIVE)) {
                actionBtnNegative = arguments.getString(PopUpActivity.ACTIONBTN_NEGATIVE)
            }
            view!!.setView(title, subTitle1, subTitle2, actionBtnPositive1,actionBtnPositive2,actionBtnPositive3, actionBtnNegative)
        }
    }

    override fun updateViewState() {}
}